### PR TITLE
refactor(components): [collapse]

### PR DESCRIPTION
**Pull request by jw-foss**
* Extract logic code out of component file.
* Extract DOM code out of component file.
* Reorganize code for better readabilities.

Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Review by chenxch — APPROVED** [review]
This code is pretty cool.

**Review by holazz — APPROVED** [review]

File: packages/components/collapse/src/use-collapse.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    emit(CHANGE_EVENT, value)
+  }
+
+  const handleItemClick = (name: CollapseActiveName) => {
+    if (props.accordion) {
+      setActiveNames([
+        (activeNames.value[0] || activeNames.value[0] === 0) &&
+        activeNames.value[0] === name
+          ? ''
+          : name,
```
Comment (btea) [inline packages/components/collapse/src/use-collapse.ts:33]: It seems that it can be simplified as:
```ts
  activeNames.value[0] === name ? '' : name
```

Comment (jw-foss) [reply packages/components/collapse/src/use-collapse.ts:33]: Right, good catch.

Comment (jw-foss) [reply packages/components/collapse/src/use-collapse.ts]: @btea updated, would you mind looking again? Thanks in advance!

**Review by btea — APPROVED** [review]

**Pull request merged**